libraries/botbuilder-ai/src/qnaMakerDialog.ts

```diff
@@ -352,8 +363,22 @@ export class QnAMakerDialog extends WaterfallDialog {
         const endpoint = {
             knowledgeBaseId: this.knowledgeBaseId,
             endpointKey: this.endpointKey,
-            host: this.hostName
+            host: this.constructHttpsHostName(this.hostName)
         };
         return new QnAMaker(endpoint);
     }
```

Comment:
@vipeketi, per our offline discussions and the internal doc, what if we look for `"qnamaker/v5.0"` in `this.hostName`?


```ts
private getQnAClient(): QnAMaker {
    const host = this.hostName.includes('qnamaker/v5.0) ?
                 this.hostName :
                 this.constructHttpsHostName(this.hostName);

    const endpoint = {
        knowledgeBaseId: this.knowledgeBaseId,
        endpointKey: this.endpointKey,
        host
    };
    return new QnAMaker(endpoint);
}
```

I'll iterate a couple of times to make the code clearer (e.g. change `constructHttpsHostName()` to `constructV4ApiHost()`)

Reply:
@vipeketi, here is the latest [commit](https://github.com/microsoft/botbuilder-js/pull/2004/commits/f5bc868bb809cab7d6edd29fbfcbfae78e2a3a8c) that supports the old behavior around constructing a v4 API endpoint while first inspecting to see if the endpoint is for the v5 API (`"qnamaker/v5"`)

Reply:
v5 handling looks good to me. When source code is downloaded from portal , the appsettings pre-populate complete url like https://xxx.azurewebsites.com/qnamaker. Can we skip just the hostname as value for this.hostName.

Reply:
We can't skip the template literal because that's a breaking change, but we can add logic to examine the provided value:

If the hostname looks like this: `xxx.azurewebsites.net`, then it's probably in the appsettings of a downloaded or deployed bot through Azure. We can apply the logic from your recent PR to the samples: https://github.com/microsoft/BotBuilder-Samples/commit/b834e98aee62dc16920d8fddfa4e8134362182bf.

If it looks like `https://xxx.azurewebsites.net/qnamaker`, then it's probably gathered from the QnAMaker portal, and it doesn't need to be touched up in this case.

Reply:
If we can accomodate support for https://xxx.azurewebsites.net/qnamaker and qnamaker/v5.0, I am good with the changes.